src/Grok.Net/Grok.cs

```diff
@@ -38,21 +39,34 @@ public Grok(string grokPattern)
         /// </summary>
         /// <param name="grokPattern">The Grok pattern to use.</param>
         /// <param name="customPatterns">A stream containing custom patterns.</param>
-        public Grok(string grokPattern, Stream customPatterns)
-            : this(grokPattern)
+        public Grok(string grokPattern, Stream customPatterns) : this(grokPattern)
         {
-            LoadCustomPatterns(customPatterns);
+            LoadCustomPatternsFromStream(customPatterns);
         }
 
         /// <summary>
         ///     Initializes a new instance of the <see cref="Grok"/> class with the specified Grok pattern and custom patterns.
         /// </summary>
         /// <param name="grokPattern">The Grok pattern to use.</param>
         /// <param name="customPatterns">Custom patterns to add.</param>
-        public Grok(string grokPattern, IDictionary<string, string> customPatterns)
-            : this(grokPattern)
+        public Grok(string grokPattern, IDictionary<string, string> customPatterns) : this(grokPattern)
         {
-            AddCustomPatterns(customPatterns);
+            LoadCustomPatternsFromDictionary(customPatterns);
+        }
+
+        /// <summary>
+        /// Initialized a new instance of the <see cref="Grok"/> class with specified Grok pattern,
+        /// custom patterns if necessary, and custom <see cref="RegexOptions"/> .
+        /// </summary>
+        /// <param name="grokPattern">The Grok pattern to use.</param>
+        /// <param name="customPatterns">Custom patterns to add (null if none are required).</param>
```

Comment:
It should not be null, if it is not required then the constructor without this param should be used.

Reply:
Sure. Wanted to avoid making multiple constructors but there's no reason to just create them. Should I then cover all cases? Right now I didn't create one that takes in `Stream` of patterns - I assume I should also add that.